src/Particle.cc

```diff
@@ -459,12 +477,63 @@ Taus::Taus(TTree* _BOOM, std::string filename, std::vector<std::string> syst_nam
   tmp= pstats["Tau2"].dmap.at("DiscrAgainstMuon");
   tau2mu= tmp;
 
-  SetBranch("Tau_idAntiEle", againstElectron);
-  SetBranch("Tau_idAntiMu",  againstMuon);
-  SetBranch("Tau_idDecayMode",  DecayMode);
+  // Get the name of the tau ID algorithm to use.
+  std::string tauIdAlgorithm(pstats["TauID"].smap.at("TauIDAlgorithm"));
```

Comment:
maybe you put it into a try catch and in the catch you take the MVA case?

Reply:
I have a new try catch block, I'm committing those changes soon.